### ISSUE TITLE
number of channels in same set should be identical  is too strict , how to modify the source code ?

### ISSUE THREAD

**Issue by CaichaoGitHub** [microsoft/nni#4372]
**Describe the issue**:
i'm tring to prune channels by  SlimPruner , but is does't support my module in torch 
the structure is a stacked Hourglass , 
i checked src from nni, i found it encounter error in ChannelMaskConflict.fixmax()
![image](https://user-images.githubusercontent.com/12059897/144852102-bd7d65db-352d-400b-9dea-96a1162c742c.png)

the network does't follow a batchnormal after conv all the time , and the channel  count will be changed after conv in the inherit,
moreover ,  the shortcut will lead to  find more predecessors . so the channel can't be same all the time .

therefore ,  i want to know how to modify the source code to adapt my network ?

the  stacked Hourglass network
![image](https://user-images.githubusercontent.com/12059897/144851804-8691409a-46f2-4060-a449-e01eeb6db476.png)

**Environment**: MAC os 
- NNI version: 2.5 
- Training service local
- Python version:3.75
- PyTorch/TensorFlow version: torch 1.9.1
- Is conda/virtualenv/venv used?: venv
- Is running in Docker?: NO

**Log message**:
 - nnimanager.log: number of channels in same set should be identical
 - dispatcher.log:
 - nnictl stdout and stderr:

**How to reproduce it?**:

**Comment by hongbinchen-njmu** [comment]
hello, have you solved this problem?
